api/classes/base_controller.js

```diff
@@ -11,133 +13,165 @@ function defaultFormatResponse(model) {
   return model;
 }
 
-function BaseController(model)  {
-  this.Model = model;
-}
+class BaseController {
+  constructor(Model) {
+    this.Model = Model;
+  }
 
-BaseController.prototype.formatRequestData = function(req) {
-  // Abstract base method,
-  // formats data for database entry
-};
+  /* eslint no-unused-vars: ["error", { "argsIgnorePattern": "^request$" }]*/
+  formatRequestData(request) {
+    // Abstract base method; formats data for database entry
```

Comment:
instead of an empty function, it might be worth having a `throw new Error("function not implemented by subclass");`